File: Solutions/readability/readability.py
import re

def main():

    text = input("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    grade = round(index)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 15:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def count_letters(text):

    letters_sum = 0;

    for i in range(len(text)):
        if re.match('[a-zA-Z]', text[i]) is not None:
            letters_sum += 1 
        else:
            continue

    return letters_sum


def count_words(text):

    words = 1;

    for i in range(len(text)):
        if re.match('[ ]', text[i]) is not None:
            words += 1
        else:
            continue

    return words


def count_sentences(text):

    sentences = 0;

    for i in range(len(text)):
        if re.match('[.!?]', text[i]) is not None:
            sentences += 1
        else:
            continue

    return sentences


main()

File: Solutions/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(char *text);
int count_words(char *text);
int count_sentences(char *text);

int main(void) 
{
	char *text = get_string("Text: ");
	// printf("%s\n", text);

	int letters = count_letters(text);
	int words = count_words(text);
	int sentences = count_sentences(text);

	double L = (double)letters / (double)words * 100;
	double S = (double)sentences / (double)words * 100;
    double index = 0.0588*L - 0.296*S - 15.8;

	int grade = (int)round(index); 

//	printf("%i letters\n", letters);
//	printf("%i words\n", words);
//	printf("%i sentences\n", sentences);

	if (grade < 1)
	{
		printf("Before Grade 1\n");
	}
	else if (grade > 15)
	{
		printf("Grade 16+\n");
	}
	else
	{
		printf("Grade %i\n", grade);
	}
	return 0;
}

int count_letters(char *text)
{
	int letters_sum = 0;

	for (int i = 0; i < strlen(text); i++)
	{
		char letter = text[i];
		if (islower(letter) || isupper(letter))
		{
			letters_sum++;
		}
	}
	
	return letters_sum;
}

int count_words(char *text)
{
	int words = 1;

	for (int i = 0; i < strlen(text); i++)
	{
		if (isspace(text[i]))
		{
			words++;
		}
	}

	return words;
}


int count_sentences(char *text)
{
	int sentences = 0;
	int p = (int)'.';

	for (int i = 0; i < strlen(text); i++)
	{
		char s = text[i];

		if (s == p && text[i+1] == p && text[i+2] == p)
		{
			sentences++;
			i += 2;
		}
		else if (s == p || s == (int)'!' || s == (int)'?')
		{
			sentences++;
		}
	}

	return sentences;
}

